### DISCUSSION TITLE
RoutedViewHost for a Stepper Component

### DISCUSSION THREAD

**Discussion by MarioXCB** [reactiveui/ReactiveUI#3723] [Q&A]
Hello,

my problem is that the Route is correctly loaded, so the content of my user control is displayed but my viewmodel in the usercontrol is empty and so no binding is happining.

I'm using the RoutedViewHost for a Stepper Component in WPF.
My Views implementing IViewFor, for example: `IViewFor<DatumStepViewModel>`

My ViewModels extending PageViewModelBase which implements the IRoutableViewModel.

My Window ViewModel implements the IScreen interface.

I'm registering the Views for the viewmodel like that
`Locator.CurrentMutable.RegisterViewsForViewModels(Assembly.GetCallingAssembly());        Locator.CurrentMutable.RegisterLazySingleton(() => new AppViewLocator(), typeof(IViewLocator));`

Also, the AppViewLocator is not called, where I have a an assignment for the viewmodel

Does anyone have a solution or an idea how I can solve this?

**Comment by ChrisPulman** [comment]
Hi, do you think it's possible for you to put an example of this together and share it so we can help to diagnose your issue.

**Reply by MarioXCB** [reply]
Sure here is an example for my stepper component: [example](https://github.com/MarioXCB/MvvmReactiveStepper)
I already found the error. I had an faulty binding in my usercontrol. 

But the custom AppViewLocator is not used, maybe you can help me with that
